quisp/modules/QRSA/HardwareMonitor/HardwareMonitor.h

```diff
@@ -51,18 +51,22 @@ class HardwareMonitor : public IHardwareMonitor {
   bool Z_Purification = false;
   int Purification_type = -1;
   int num_measure;
+  int num_end_nodes;
+  std::vector<int> tomography_partners;
+  // in the case of retry connection setup, the partner could be changed.
+  std::multimap<int, int> qnic_partner_map;
```

Comment:
why do you need to use `std::multimap`? why not `std::map` ?

Reply:
This is for future implementation. Currently, the qnic address (key)  should have a single partner address (value). However, in the regime of multiplexing, one qnic to several partners can be possible.
But now I realized this could be achievable with just map. So let me use map instead.

Reply:
IMO, multimap is not suitable for multiplexing. should be `std::map<QNIC index, std::vector<Partner addr>>`. or create a new class to manage it. anyway, let's think about it later.